### PR TITLE
OCPQE-12319: Migrate pipeline into Prow CI --128 profile

### PR DESCRIPTION
**Pull request by LiangquanLi930**
[OCPQE-12319](https://issues.redhat.com//browse/OCPQE-12319): Migrate pipeline into Prow CI --128 profile

### CODE REVIEW

**Comment by heliubj18** [comment]
@kuiwang02 is it ok to run so many cases in ginkgo & cucushift for almost all the components ? In fact we only need a smallest set of testcases.  This test is for management cluster, no relation with hypershift hosted cluster.

**Comment by heliubj18** [comment]
LGTM except for my question to @kuiwang02.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml [hunk 1]
```diff
@@ -99,6 +99,18 @@ tests:
     test:
     - chain: cucushift-sdn-migration-ovn
     workflow: cucushift-installer-rehearse-aws-ipi
+- as: e2e-aws-ipi-ovn-hypershift-p2
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:102]: We are migrating tests from Jenkins into Prow, and the related profile are retired then in Jenkins.
Currently we are doing this for 4.12 (migrate one profile to Prow, then retire the profile in Jenkins).
We haven't started the process for 4.11 yet (We plan to do that after we are done for 4.12), we should not add it for 4.11 at the moment so that we have less duplicate tests exist in both Jenkins and Prow (there are 4.11 tests exist in Prow because they are added before we start 4.12).

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml [hunk 2]
```diff
@@ -129,6 +129,18 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8
+- as: e2e-aws-ipi-ovn-hypershift-p2
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:132]: BTW, we already have `e2e-aws-ipi-hypershift-p2` in 4.12.
Since OVN is the default in 4.12, `e2e-aws-ipi-ovn-hypershift-p2` and `e2e-aws-ipi-hypershift-p2` are duplicate.

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:132]: `e2e-aws-aws-ipi-hypershift-p2` no longer exists, we changed `e2e-aws-ipi-ovn-hypershift-mgmt-p2` for better understanding

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:132]: We use the same workflow `cucushift-installer-rehearse-aws-ipi-ovn-hypershift` to set up two clusters, and use cluster1 to test test-suite1, and use cluster2 to test test-suite2. This waste of resources and time (duplicate installation).
Is this the direction we are going ? @jhou1 @chengzhang1016

**Comment by kuiwang02** [comment]
> @kuiwang02 is it ok to run so many cases in ginkgo & cucushift for almost all the components ? In fact we only need a smallest set of testcases. This test is for management cluster, no relation with hypershift hosted cluster.
> 
> Is there any tags/labels to help reduce number of cases ?

For me I do not think you need this test.
you could change one existing test to change your workflow cucushift-installer-rehearse-aws-ipi-ovn-hypershift
for example , because your workflow is based on cucushift-installer-rehearse-aws-ipi-ovn, so you could change existing test
```yaml
- as: e2e-aws-ipi-ovn-p3
  cron: 0 21 */28 * *
  steps:
    cluster_profile: aws-qe
    env:
      BASE_DOMAIN: qe.devcluster.openshift.com
      E2E_RUN_TAGS: '@aws-ipi and @network-ovnkubernetes and not @fips'
      TAG_VERSION: '@4.12'
    test:
    - chain: openshift-e2e-test-qe
    workflow: cucushift-installer-rehearse-aws-ipi-ovn
```
to
```yaml
- as: e2e-aws-ipi-ovn-hypershift-p3
  cron: 0 21 */28 * *
  steps:
    cluster_profile: aws-qe
    env:
      BASE_DOMAIN: qe.devcluster.openshift.com
      E2E_RUN_TAGS: '@aws-ipi and @network-ovnkubernetes and not @fips'
      TAG_VERSION: '@4.12'
    test:
    - chain: openshift-e2e-test-qe
    workflow: cucushift-installer-rehearse-aws-ipi-ovn-hypershift
```
no need new testing.

your current way will take more resource which is not necessary.

**Comment by kuiwang02** [comment]
if you still want to use cucushift-installer-rehearse-aws-ipi-ovn-hypershift and create new testing, you could use TEST_IMPORTANCE to select high severity case, like Critial, which is small case set.
like
```yaml
TEST_IMPORTANCE: Critical
```

**Comment by YuLi517** [comment]
@kuiwang02 @liangxia Even if we just run test on management cluster, we still need to create Hosted cluster. When we installed hosted cluster, the management cluster will get new pods added for hypershift purpose. On this env, we can say it's a entire hypershift cluster which can be used by real customer. Our e2e test should run on this kind of env to prove whether real customer can really use our hypershift. On management cluster, we will run 2 part of e2e: 1) OCP sub component e2e cases;2) hypershift operator related new function e2e test. 

In general, we need to run e2e on both management cluster and hosted cluster. In addition, these 2 cluster will use different kubeconfig.

**Comment by kuiwang02** [comment]
> @kuiwang02 @liangxia Even if we just run test on management cluster, we still need to create Hosted cluster. When we installed hosted cluster, the management cluster will get new pods added for hypershift purpose. On this env, we can say it's a entire hypershift cluster which can be used by real customer. Our e2e test should run on this kind of env to prove whether real customer can really use our hypershift. On management cluster, we will run 2 part of e2e: 1) OCP sub component e2e cases;2) hypershift operator related new function e2e test.
> 
> In general, we need to run e2e on both management cluster and hosted cluster. In addition, these 2 cluster will use different kubeconfig.

so, as https://github.com/openshift/release/pull/33023#issuecomment-1275479325 said, you have to create such test with this workflow, and then you could use TEST_IMPORTANCE for golang to select small case set as requested by @heliubj18

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml (hunk 2 above)
Comment (heliubj18) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:132]: @kuiwang02 I see both you have the comments to merge into one tests based on this workflow. I prefer to keep them all for different test scenarios.
1) The previously `e2e-aws-ipi-ovn-hypershift-mgmt-p2` is use to test hypershift hosted cluster including control plane and worker nodes.  This testsuit  includes hypershift core functions testcases together with OCP e2e on hosted clusters.  It is valuable to let us know if there is any error/bug in hypershift side for self-managed and even managed hypershift service.

2) The new added test in this PR is based on the same workflow. But it is for the work moving jenkins CI to prow. That is to replace [128_IPI on AWS & OVN & HyperShift](https://gitlab.cee.redhat.com/aosqe/ci-profiles/-/blob/master/4.12/128_IPI%20on%20AWS%20%26%20OVN%20%26%20HyperShift.install.yaml).
This CI is used to test management cluster instead of hypershift hosted clusters. It is a general test for OCP component e2e just to make sure the management cluster should not be affected by hypershift. Because hypershift control plane pods & operator pods are just applications in the mgmt cluster.
This scenario may include many testcases.  

In short, the first one is for hypershift hosted cluster CI. It is used to check HyperShift core functions & the hosted cluster functions are as expected
Another is for OCP CI with hypershift installed. It is used to check OCP itself before a new version of OCP is released.

Comment (kuiwang02) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:132]: @heliubj18 I understand your reply is similar with https://github.com/openshift/release/pull/33023#issuecomment-1275494495, and my reply is https://github.com/openshift/release/pull/33023#issuecomment-1275497716

**Comment by heliubj18** [comment]
> 

@kuiwang02  Thanks a lot! `TEST_IMPORTANCE` is just what I want.

**Comment by LiangquanLi930** [comment]
@kuiwang02 could you help to review again?

**Comment by kuiwang02** [comment]
i am good.

**Comment by LiangquanLi930** [comment]
@liangxia could you help me review this PR again? thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml [hunk 3]
```diff
@@ -129,6 +129,19 @@ tests:
...
+  cron: 0 4 */7 * *
+  steps:
+    cluster_profile: aws-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      E2E_RUN_TAGS: '@aws-ipi and @network-ovnkubernetes and not @proxy and not @disconnected
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:138]: No need `not @proxy and not @disconnected`.
E.g, for a test case only works on disconnected cluster, it has disconnected tag.
if a test case only works on connected cluster, it has connected tag.
if the test cases works on both connected and disconnected cluster, then it has both connected tag and disconnected tag.
`not @disconnected` will exclude those test cases that can work on both connected cluster and disconnected cluster.
